lib/shelf-reference.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+module.exports = (logger, host, requestPromise, ShelfFile, utils) => {
+    /**
+     * A reference to a Shelf bucket.
+     */
+    class ShelfReference {
+        /**
+         * @param {string} refName
+         * @param {string} authToken
+         */
+        constructor(refName, authToken) {
+            this.authToken = authToken;
+            this.refName = refName;
+        }
+
+
+        /**
+         * Creates and returns a ShelfFile at the given path.
+         *
+         * @param {string} path
+         * @param {Boolean} unique Denotes if the uploaded content should have
+         *                         a unique url.
+         * @return {ShelfFile}
+         */
+        getArtifact(path, unique) {
+            var uri;
+
+            logger.debug("In ShelfReference.getArtifact()");
+            uri = utils.buildUrl(path, host, this.refName, unique);
+
+            return new ShelfFile(path, host, uri, this.authToken);
```

Comment:
Yeah, this does seem off. I'll rework it. I didn't know that about URIjs though. I should have looked at it more.